server/applications.json

```diff
@@ -1302,6 +1302,31 @@
                 }
             ]
         },
+        "shell": {
```

Comment:
Why there is setting for this?

Reply:
Thanks. @m-u-r-p-h-y @dee-ynput your thoughts?

I still feel, based on the comments above by everyone, we will need a way to allow the user to pick what shell to trigger that works for their needs.

Also, the assumption that this PR may solve the client's needs based on their ticket @m-u-r-p-h-y will still need to be confirmed. That may also clarify any customization needs as well. So may be good to follow up on that before we continue here?

---

Side note: I do know that @Lypsolon also requested this feature to be able to quickly access a terminal in the environment of the host - so it would maybe still solve at least a request from him.

Reply:
We should use SHELL on linux and mac. On windows`wt.exe` if exists otherwise `cmd.exe` and you have to know which is used so you can e.g. add  `/K` for cmd.

Reply:
Side note: For me `wt.exe` actually didn't work even though I do have the executable. I was unable to run it with an initialized environment it seemed - it'd always use the system environment and also ignore `cwd`.

Reply:
> I still feel, based on the comments above by everyone, we will need a way to allow the user to pick what shell to trigger that works for their needs.

let share my user story, I hope it helps: 

Personally, I'd expect this to exist in studio site settings or application settings.
`studio site settings` allows me to have my own customization (which is none atm) at the cost of doing this for every machine I use to log into the system. as I user I can have multiple site ids.
`application settings` is pretty straight forward. it also allows me to open a console in some host specific. which I may need for testing purposes. 
e.g. in shell application(currently launched via the debug shell), I'm able to access ayon libs!  e.g. `ayon_core`
![image](https://github.com/user-attachments/assets/879cc181-b42e-4043-89c5-2b488066ffb6)


Regarding the customization per users, on my side, I'm just a regular guy, who have a basic setup to get the job done.
I'd either go with `cmd` or `pwsh` (we don't need to specify the whole path in the shell app settings because I already added it to PATH on my machine)
Also, I don't have any customization therefore, shell application is fine.

Reply:
> Why there is setting for this? Please don't, it's not used as application, you want to run other application with shell.

Yes it is an application.
It just happens to be an application which **can** run other applications.
Can you elaborate on why we shoud not do that @iLLiCiTiT ?

I have a feeling there's a miss understanding of the purpose. Maybe because of different meaning we have for the words `shell`, `terminal`, `console` ?

The goal is to open a console where the user can do what they want.
(Maybe you're thinking about launching eg Maya and have a terminal showing up to see log and such?)

> We should use SHELL on linux and mac.

Yes, `$SHELL` is probably the default value for linux and mac.
But the studio and the projects must be able to configure something else, and variants too.

Guys, let's ship this, get some feedback, and then come back to the discussion.

Reply:
@MustafaJafar I've fixed the regular app launch with this https://github.com/ynput/ayon-applications/pull/12/commits/144163d05a2a714f11e807fa0d29d3d7feca88c8 comit.

That commit should also fix launch on Linux and Max because apparently `subprocess.CREATE_NEW_CONSOLE` does not exist on Linux. It's windows only?

@Lypsolon would be great if you could just give this a Linux test-drive.
And I'm clueless as to who would be capable of testing Mac.

Reply:
It is not application like maya, nuke or something like that. We've already removed shell from setting once because it raised exactly the issues we're mentioning here. The terminal cannot be set "universaly" as it is machine specific. Also it doesn't make sense to have variants as we have it.

We should have auto-detect logic, and maybe site settings for shell, but it is 100% not the same application as we have the others.

We need shell for running specific action, that will start terminal window which has set environment as other application, not run "shell as application".

Reply:
@dee-ynput @antirotor Please provide your opinions so we can **force** this into **removal of terminal/shell** from application settings OR autodetect it in a different way.

I suspect there's _still_ merit to just a 'shallow' shell within the AYON environment without the context of other applications which may warrant exposing a "shell" application but I'm not convinced either. Please take into account @iLLiCiTiT 's points above ☝️ and let's get this ball rolling.

Reply:
> I think that having shell as "standard" application in AYON might be useful,

I really don't see any usecase when the terminal application would be used as "application" and would be useful. I would say that the action added in this PR is useful for developer, but terminal application is not.

If it would be for development, you can easily add it in additional applications. But I would not put it to settings to make it visible in applications enumerator.

> We need default/studio/project definition of what the shells are so we have some sane starting point. This should be defined as standard application in the application addon with variants and everything as was already shown in this discussion - it allows running the shell console in selected context.

I've mentioned it already but I'm strongly against having it in `Applications` settings.

Even this PR is not using the shell application as application, but is adding another action, different from application launch action to launch the terminal.

Ad terminal choice, on macOs it is `Terminal.app` (probably no other choice?). On windows it can be powershell 7, powershell 6, cmd, git console, on linux it can be many many applications. On linux you probably should know which exactly terminal it is so you know how to work with it and start the session correctly. Meaning it is not really studio specific but machine specific. By default we should do as much as we can to start correct console, but only if user don't like it, he should set it in site settings, but not in studio settings.

Right now I'm probably only one who thinks this is bad idea, so I'll probably loose this fight, just wanted to mention this (sorry).